db/dataset.py

```diff
@@ -192,3 +193,116 @@ def delete(id):
     """Delete dataset with a specified ID."""
     with db.engine.begin() as connection:
         connection.execute("DELETE FROM dataset WHERE id = %s", (str(id),))
+
+
+def create_snapshot(dataset_id):
+    """Creates a snapshot of current version of a dataset.
+
+    Snapshots are stored as JSON and have the following structure:
+    {
+        "name": "..",
+        "description": "..",
+        "classes": [
+            {
+                "name": "..",
+                "description: "..",
+                "recordings": ["..", ...]
+            },
+            ...
+        ]
+    }
+
+    Args:
+        dataset_id (string/uuid): ID of a dataset.
+
+    Returns:
+        ID (UUID) of a snapshot that was created.
+    """
+    dataset = get(dataset_id)
+    if not dataset:
+        raise exceptions.NoDataFoundException("Can't find dataset with a specified ID.")
+    snapshot = {
+        "name": dataset["name"],
+        "description": dataset["description"],
+        "classes": dataset["classes"],
+    }
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            INSERT INTO dataset_snapshot (id, dataset_id, data)
+                 VALUES (uuid_generate_v4(), :dataset_id, :data)
+              RETURNING id
+        """), {
+            "dataset_id": dataset_id,
+            "data": json.dumps(snapshot),
+        })
+        return result.fetchone()["id"]
+
+
+def get_snapshot(id):
+    """Get snapshot of a dataset.
+
+    Args:
+        id (string/uuid): ID of a snapshot.
+
+    Returns:
+        dictionary: {
+            "id": <ID of the snapshot>,
+            "dataset_id": <ID of the dataset that this snapshot is associated with>,
+            "created": <creation time>,
+            "data": <actual content of a snapshot (see `create_snapshot` function)>
+        }
+    """
+    with db.engine.connect() as connection:
+        result = connection.execute(sqlalchemy.text("""
+            SELECT id::text
+                 , dataset_id::text
+                 , data
+                 , created
+              FROM dataset_snapshot
+             WHERE id = :id
+        """), {"id": id})
+        row = result.fetchone()
+        if not row:
+            raise db.exceptions.NoDataFoundException("Can't find dataset snapshot with a specified ID.")
+        return dict(row)
+
+def delete_snapshot(connection, snapshot_id):
+    """Delete a snapshot.
+
+    Args:
+        connection: an SQLAlchemy connection.
+        snapshot_id (string/uuid): ID of a snapshot.
+    """
+    query = sqlalchemy.text("""
+        DELETE FROM dataset_snapshot
+              WHERE id = :snapshot_id""")
+    connection.execute(query, {"snapshot_id": snapshot_id})
+
+def get_snapshots_for_dataset(dataset_id):
```

Comment:
I'm not sure if it's a good idea to add stuff that is unused. Makes things even more confusing. If something changes we'll have to spend time updating these functions that are just ran by tests. And tests too, potentially.